compiler/modules/front/codeGen.bal

```diff
@@ -495,12 +497,19 @@ function codeGenWhileStmt(CodeGenContext cx, bir:BasicBlock startBlock, Environm
     }
 }
 
-function validLoopAssignments(CodeGenContext cx, int[] assignments) returns CodeGenError? {
+function validLoopAssignments(CodeGenContext cx, Binding? binding, int[] assignments) returns CodeGenError? {
+    // pr-todo: I think swapping the loop order will be more efficient
     foreach int r in assignments {
-        if r < cx.loopStartRegister() {
```

Comment:
I think this means we are not using loopStartRegister anywhere.

Reply:
Yes, I removed it but have pushed to the wrong branch, I have accidentally opened this with my tmp branch.